assets/js/base/components/price-slider/index.js

```diff
@@ -0,0 +1,405 @@
+/**
+ * External dependencies
+ */
+import { sprintf, __ } from '@wordpress/i18n';
+import { Fragment, useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { useDebounce } from '@woocommerce/base-hooks';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { constrainRangeSliderValues, formatCurrencyForInput } from './utils';
+
+const PriceSlider = ( {
+	initialMin,
+	initialMax,
+	min,
+	max,
+	onChange,
+	step,
+	currencySymbol,
+	priceFormat,
+	showInputFields,
+	showFilterButton,
+	isLoading,
+} ) => {
+	const [ value, setValue ] = useState( {
+		min: initialMin,
+		max: initialMax,
+	} );
+	const [ formattedValue, setFormattedValue ] = useState( {
+		min: formatCurrencyForInput( value.min, priceFormat, currencySymbol ),
+		max: formatCurrencyForInput( value.max, priceFormat, currencySymbol ),
+	} );
+	const debouncedChangeValue = useDebounce( value, 500 );
+
+	useEffect( () => {
+		if (
+			value.min === undefined ||
+			value.max === undefined ||
+			min > value.min ||
+			max < value.max
+		) {
+			setValue( {
+				min,
+				max,
+			} );
+		}
+		setFormattedValue( {
+			min: formatCurrencyForInput(
+				value.min,
+				priceFormat,
+				currencySymbol
+			),
+			max: formatCurrencyForInput(
+				value.max,
+				priceFormat,
+				currencySymbol
+			),
+		} );
+	}, [ min, max ] );
+
+	useEffect( () => {
+		setFormattedValue( {
+			min: formatCurrencyForInput(
+				value.min,
+				priceFormat,
+				currencySymbol
+			),
+			max: formatCurrencyForInput(
+				value.max,
+				priceFormat,
+				currencySymbol
+			),
+		} );
+	}, [ value ] );
+
+	useEffect( () => {
+		if ( ! showFilterButton && ! isLoading ) {
+			triggerChangeCallback();
+		}
+	}, [ debouncedChangeValue ] );
+
+	const triggerChangeCallback = () => {
+		onChange( value );
+	};
+
+	/**
+	 * Handles styles for the shaded area of the range slider.
+	 */
+	const getProgressStyle = () => {
+		const low =
+			Math.round( 100 * ( ( value.min - min ) / ( max - min ) ) ) - 0.5;
+		const high =
+			Math.round( 100 * ( ( value.max - min ) / ( max - min ) ) ) + 0.5;
+
+		return {
+			'--low': low + '%',
+			'--high': high + '%',
+		};
+	};
+
+	/**
+	 * Works around an IE issue where only one range selector is visible by changing the display order
+	 * based on the mouse position.
+	 *
+	 * @param {obj} event event data.
+	 */
+	const findClosestRange = ( event ) => {
```

Comment:
This should be wrapped in `useCallback` along with the appropriate dependencies because it is an event handler passed through to a component.